foundation/foundation-archetype/src/main/resources/archetype-resources/deploy.sh

```diff
@@ -5,31 +5,33 @@ INFRA_NAME=infrastructure
 
 print_usage() {
   echo "Usage: $0 [up|down|shutdown]"
-  echo "   up          deploy application (automatically starts deployment infrastructure if needed)"
-  echo "   down        tear down application"
-  echo "   shutdown    tear down deployment infrastructure"
+  echo "   startup      create/upgrade deployment infrastructure"
+  echo "   shutdown     tear down deployment infrastructure (tears down application if needed)"
+  echo "   up           deploy application (starts deployment infrastructure if needed)"
+  echo "   down         tear down application"
 }
 
 startup() {
-  echo "Checking for deployment infrastructure..."
-  helm status $INFRA_NAME > /dev/null 2>&1
-  if [ $? -ne 0 ]; then
-    helm upgrade --install $INFRA_NAME ${artifactId}-infrastructure \
-      --values ${artifactId}-infrastructure/values.yaml \
-      --values ${artifactId}-infrastructure/values-dev.yaml
-    if ! kubectl wait -A --for=condition=Ready pod -l app.kubernetes.io/name=argocd-server; then
-      exit $?
-    fi
-    argocd repo add ${projectGitUrl} --server localhost:30080 --plaintext --insecure-skip-server-verification
+  echo "Deploying infrastructure..."
+  helm upgrade --install $INFRA_NAME ${artifactId}-infrastructure \
+     --values ${artifactId}-infrastructure/values.yaml \
+     --values ${artifactId}-infrastructure/values-dev.yaml
+  if ! kubectl rollout status --namespace argocd deployment/argocd-server --timeout=30s; then
+    exit $?
   fi
+  argocd repo add ${projectGitUrl} --server localhost:30080 --plaintext --insecure-skip-server-verification
 }
 
 is_app_running() {
   argocd app get $APP_NAME --server localhost:30080 --plaintext > /dev/null 2>&1
 }
 
 deploy() {
-  startup
+  echo "Checking for deployment infrastructure..."
+  helm status $INFRA_NAME > /dev/null
+  if [ $? -ne 0 ]; then
```

Comment:
I: Move existence check to `deploy` so we can upgrade the infra chart with `startup` if needed